RevenueCatUI/Data/PaywallViewConfiguration.swift

```diff
@@ -97,4 +97,22 @@ extension PaywallViewConfiguration.Content {
         return offering.map(Self.offering) ?? .defaultOffering
     }
 
+    /// Resolves the content to an `Offering` by fetching from the backend if needed.
+    /// - Returns: The resolved `Offering`, or `nil` if it couldn't be fetched.
+    func resolveOffering() async -> Offering? {
+        do {
+            switch self {
+            case let .offering(offering):
+                return offering
+            case .defaultOffering:
+                return try await Purchases.shared.offerings().current
```

Comment:
Ideally we should avoid using `Purchases.shared` in RevenueCatUI and all interactions with it should go through `PaywallPurchasesType`. That helps the RC app work correctly with previews.

For this case, that could be achieved passing the `PurchaseHandler` as a parameter and adding a `func offerings()` bridge method to it.